sc-kpm/sc-agents-common/agents/GenerationByTemplateAgent.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2022 Intelligent Semantic Systems LLC, All rights reserved.
+ */
+
+#include "sc-memory/scs/scs_types.hpp"
+
+#include "utils/CommonUtils.hpp"
+#include "utils/IteratorUtils.hpp"
+
+#include "GenerationByTemplateAgent.hpp"
+
+namespace scAgentsCommon
+{
+SC_AGENT_IMPLEMENTATION(GenerationByTemplateAgent)
+{
+  ScAddr questionNode = m_memoryCtx.GetEdgeTarget(edgeAddr);
+  if (!checkActionClass(questionNode))
+    return SC_RESULT_OK;
+
+  SC_LOG_DEBUG("GenerationByTemplateAgent started")
+
+  ScAddr templateLinkAddr = utils::IteratorUtils::getAnyByOutRelation(&m_memoryCtx, questionNode, CoreKeynodes::rrel_1);
```

Comment:
This is link with template system id? Can we cant just pass here template itself?